### PR TITLE
Output Management

### LINKED ISSUE #949: Guides - Managing Built Files
> aka "intro to why you should use html-webpack-plugin"

Based on feedback from [twitter](https://twitter.com/karljakoblind/status/834820883415961600):

> the new docs. Main confusion for me was i included generated files in wrong order and got not helpful error msg

Few brainstorming options but I think that we should essentially recommend heavily the use of html-webpack-plugin and why. (Explain the whole managing new build files that are hashed every time, ensuring they are output in correct order, etc)

### PR DESCRIPTION
**Pull request by skipjack**
Add a best practices and advice guide for managing built files (aka webpack's output).

Resolves #949

### CODE REVIEW

**Comment by skipjack** [comment]
@TheLarkInn got a rough start going, let me know what you think.

File: content/guides/output-management.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: Output Management
+contributors:
+  - skipjack
+---
+
+Managing webpack's [output]() and including it in your HTML files may not seem tough at first, but once you start [using hashes in filenames]() and outputting [multiple bundles](), things can start to get a bit hairy. However, there's no need to fear as a few plugins exist that will make this process much easier to manage.
```
Comment (simon04) [inline content/guides/output-management.md]: Here, three links are missing (`()` in the markdown).

File: content/guides/output-management.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    <link rel="stylesheet" href="/styles.min.css" />
+    <script src="/vendor.bundle.js"></script>
+  </head>
+  <body>
+    <script src="/app.bundle.js"></script>
+  </body>
+</html>
+```
+
+Here we've loaded a favicon, our stylesheet (extracted with the [extract-text-webpack-plugin](/plugins/extract-text-webpack-plugin)), any libraries (split into [a separate bundle](/guides/code-splitting-libraries)), and finally our main bundle (`app.bundle.js`). This is ok, but what happens if we change our entry point names? What if we decide to take advantage of [better caching](/guides/caching)?
```
Comment (simon04) [inline content/guides/output-management.md]: According to https://webpack.js.org/writers-guide/#typesetting, plugins are enclosed in backticks and camel-cased, such as `ExtractTextWebpackPlugin`.

File: content/guides/output-management.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+---
+title: Output Management
```
Comment (simon04) [inline content/guides/output-management.md]: Maybe it's just me, but I have some troubles guessing the content from this title. "Building index.html" or "Generating index.html" disregards the last two sections, but might be easier to grasp. Any thoughts?

**Comment by simon04** [comment]
Btw: https://github.com/webpack/webpack/issues/1905#issuecomment-276184779 mentions yet another option, [`HtmlPlugin.js`](https://gist.github.com/MagicDuck/f57ea85bc859d873158c5a8bc12b7466). Not sure whether there's a place for that in this guide.

File: content/guides/output-management.md (hunk 3 above)
Comment (skipjack) [inline content/guides/output-management.md]: Hmm yeah, I see what you mean but I wanted to keep it a bit more general than that. Maybe __Output and Bundle Management__?

**Comment by skipjack** [comment]
> Btw: webpack/webpack#1905 (comment) mentions yet another option, HtmlPlugin.js. Not sure whether there's a place for that in this guide.

Hmmm yeah, I'm not sure. It doesn't seem like it's been used very much (or available on npm) but maybe it would fit in a `## Other Options` or `## Related` section of links at the bottom of the page?

File: content/guides/output-management.md (hunk 3 above)
Comment (skipjack) [inline content/guides/output-management.md]: Or... how about the one from #949 -- __Managing Built Files__ or __Managing Output Files__?

I think one of those would be an improvement without locking it in to an html-only discussion.

**Comment by skipjack** [comment]
@TheLarkInn merged this but feel free to follow up with any edits or to elaborate on __The Manifest__ section.

**Pull request merged**